signal/reporter_test.go

```diff
@@ -1,3 +1,5 @@
+// +build unit
```

Comment:
Why do we need this?

Reply:
Build tags tell go test which tests to include in a test run. When you run `go test`, it won't include `+build integration`, but will run any tests that are not tagged. This isn't really what I wanted, since it basically means you can either run unit tests, or unit tests _and_ integration tests.

I think it's better to be explicit and say run unit tests _or_ integration tests. To do that, all your unit tests need to be tagged too.

The downside is that `go test ./...` no longer runs any tests by default. But tests are usually run by Make anyway, and I've  added targets for unit and integration tests.

With make, we can run:

`make unit`: run just the unit tests
`make integration`: run just the integration tests
`make test`: run all tests